process/peer/process.go

```diff
@@ -336,6 +337,112 @@ func (vs *validatorStatistics) RootHash() ([]byte, error) {
 	return vs.peerAdapter.RootHash()
 }
 
+func (vs *validatorStatistics) getValidatorDataFromLeaves(
+	leaves map[string][]byte,
+) (map[uint32][]*state.ValidatorInfoData, error) {
+
+	validators := make(map[uint32][]*state.ValidatorInfoData, vs.shardCoordinator.NumberOfShards()+1)
+	for i := uint32(0); i < vs.shardCoordinator.NumberOfShards(); i++ {
+		validators[i] = make([]*state.ValidatorInfoData, 0)
+	}
+	validators[core.MetachainShardId] = make([]*state.ValidatorInfoData, 0)
+
+	sortedLeaves := make([][]byte, len(leaves))
+	i := 0
+	for _, pa := range leaves {
+		sortedLeaves[i] = pa
+		i++
+	}
+
+	sort.Slice(sortedLeaves, func(i, j int) bool {
+		return bytes.Compare(sortedLeaves[i], sortedLeaves[j]) < 0
+	})
+
+	for _, pa := range sortedLeaves {
+		peerAccount := &state.PeerAccount{}
```

Comment:
might consider extracting this in a function call unmarshalPeer

Reply:
fixed